components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/graphQL/GraphQLAPIHandler.java

```diff
@@ -0,0 +1,296 @@
+            String operationList = "";
+            Parser parser = new Parser();
+
+            ArrayList<String> operationArray = new ArrayList<>();
+
+            org.apache.axis2.context.MessageContext axis2MC = ((Axis2MessageContext) messageContext).
+                    getAxis2MessageContext();
+            String requestPath = ((Axis2MessageContext) messageContext).getProperties().
+                    get(REST_SUB_REQUEST_PATH).toString();
+            if (!requestPath.isEmpty()) {
+                String[] queryParams = ((Axis2MessageContext) messageContext).getProperties().
+                        get(REST_SUB_REQUEST_PATH).toString().split(QUERY_PATH_STRING);
+                if (queryParams.length > 1) {
+                    String queryURLValue = queryParams[1];
+                    payload = URLDecoder.decode(queryURLValue, UNICODE_TRANSFORMATION_FORMAT);
+                } else {
+                    RelayUtils.buildMessage(axis2MC);
+                    OMElement body = axis2MC.getEnvelope().getBody().getFirstElement();
+                    if (body != null && body.getFirstElement() != null) {
+                        payload = body.getFirstElement().getText();
+                    } else {
+                        log.debug("Invalid query parameter " + queryParams[0]);
+                        handleFailure(messageContext, "Invalid query parameter");
+                        return false;
+                    }
+                }
+            } else {
+                handleFailure(messageContext, "Request path cannot be empty");
+                return false;
+            }
+
+            // Validate payload with graphQLSchema
+            Document document = parser.parseDocument(payload);
+
+            if (validatePayloadWithSchema(messageContext, document)) {
+                supportForBasicAndAuthentication(messageContext);
+                // Extract the operation type and operations from the payload
+                for (Definition definition : document.getDefinitions()) {
+                    if (definition instanceof OperationDefinition) {
+                        OperationDefinition operation = (OperationDefinition) definition;
+                        if (operation.getOperation() != null) {
+                            ((Axis2MessageContext) messageContext).getAxis2MessageContext().
+                                    setProperty(Constants.Configuration.HTTP_METHOD, operation.getOperation().toString());
+                            if (Operation.QUERY.equals(operation.getOperation())) {
+                                for (Selection selection : operation.getSelectionSet().getSelections()) {
+                                    if (selection instanceof Field) {
+                                        Field field = (Field) selection;
+                                        operationArray.add(field.getName());
+                                        log.debug("Operation - Query " + field.getName());
```

Comment:
check for isDebugEnabled if you do string concats

Reply:
Fixed with 2a6fbdd3ac9ce54ca7bc94a45328a41826836a80